File: 프로그래머스/2/42885. 구명보트/구명보트.js
function solution(people, limit) {
    let boat = 0;
    let first = 0;
    let last = people.length-1
    people.sort((a,b) => a - b)
    
    while (first <= last) {
        if (people[first] + people[last] <= limit) first++;
        last--;
        boat++;
    }
    
    return boat;
}

File: 프로그래머스/2/42885. 구명보트/구명보트.py
def solution(people, limit):
    people.sort()
    boat = 0
    i = 0
    j = len(people) - 1
    
    while i <= j:
        if people[i] + people[j] <= limit: i += 1
        j -= 1
        boat += 1
    
    return boat